Bootstrapper/appleGuiceBootstrapper.cpp

```diff
@@ -198,73 +200,83 @@ void parseLine(string &headerEntry,
 	}
 }
 
+set<string> filterSuperProtocols(set<string> superProtocols) {
+    set<string> filteredSuperProtocols;
+    
+    for(set<string>::const_iterator protocolNameIterator = superProtocols.begin(); protocolNameIterator != superProtocols.end(); protocolNameIterator++ ) {
+        string protocolName = *protocolNameIterator;
+        
+        if (!isNSObject(protocolName)) {
```

Comment:
since you are iterating over a set, `"NSObject"` will appear either 0 or 1 times in it. You can just check if the set contains "NSObject" and if so remove it.

Reply:
I've read this thread - https://stackoverflow.com/questions/5838922/how-to-delete-an-object-in-a-set
Do you think we can use 'erase'?

Assuming you're calling the set's erase() method, note that this will call the destructor of the object for you. After you erase() your object, it has already been deleted, and thus your second attempt to manually call delete will fail as the pointer is no longer allocated.

Reply:
using erase is cool, but it seems you don't even call `filterSuperProtocols` so you might want to erase the whole thing.

Reply:
Ok deleted it